src/Internal/Logger.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * Class Logger
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal;
+
+use WC_Logger;
+use WC_Payments;
+use WCPay\Core\Mode;
+use WCPay\Internal\Proxy\LegacyProxy;
+
+defined( 'ABSPATH' ) || exit; // block direct access.
+
+/**
+ * A wrapper class for interacting with WC_Logger.
+ */
+class Logger {
+
+	const LOG_FILENAME = 'woocommerce-payments';
+
+	/**
+	 * The holding property for our WC_Logger instance.
+	 *
+	 * @var WC_Logger $logger
+	 */
+	private $logger;
+
+	/**
+	 * Legacy proxy.
+	 *
+	 * @var LegacyProxy
+	 */
+	private $legacy_proxy;
+
+	/**
+	 * Mode
+	 *
+	 * @var Mode
+	 */
+	private $mode;
+
+	/**
+	 * Logger constructor.
+	 *
+	 * @param LegacyProxy $legacy_proxy    Legacy proxy.
+	 * @param string      $logger_function Logger function.
+	 * @param Mode        $mode            Mode.
+	 */
+	public function __construct( LegacyProxy $legacy_proxy, string $logger_function, Mode $mode ) {
+		$this->legacy_proxy = $legacy_proxy;
+		$this->logger       = $legacy_proxy->call_function( $logger_function );
+		$this->mode         = $mode;
+	}
+
+	/**
+	 * Add a log entry.
+	 *
+	 * Note that this depends on WC_Payments gateway property to be initialized as
+	 * we need this to access the plugins debug setting to figure out if the setting
+	 * is turned on.
+	 *
+	 * @param string $message Log message.
+	 * @param string $level One of the following:
+	 *     'emergency': System is unusable.
+	 *     'alert': Action must be taken immediately.
+	 *     'critical': Critical conditions.
+	 *     'error': Error conditions.
+	 *     'warning': Warning conditions.
+	 *     'notice': Normal but significant condition.
+	 *     'info': Informational messages.
+	 *     'debug': Debug-level messages.
+	 * @throws \Exception Throws Exception if log cannot be written.
+	 */
+	public function log( $message, $level = 'info' ) {
+		if ( ! $this->can_log() ) {
+			return;
+		}
+
+		$this->logger->log( $level, $message, [ 'source' => self::LOG_FILENAME ] );
+	}
+
+
+	/**
+	 * Checks if the gateway setting logging toggle is enabled.
+	 *
+	 * @return bool Depending on the enable_logging setting.
+	 * @throws \Exception Throws Exception for any issue in accessing mode.
+	 */
+	public function can_log() {
+
+		if ( $this->mode->is_dev() ) {
+			return true;
+		}
+
+		return 'yes' === WC_Payments::get_gateway()->get_option( 'enable_logging' );
```

Comment:
My previous comment (https://github.com/Automattic/woocommerce-payments/pull/7462#discussion_r1356331935) still applies: `WC_Payment_Gateway_WCPay` can be a dependency of this class, and would not need to be retrieved statically.

FWIW, you'd find this out when you're trying to reach 100% `src` coverage anyway.